docs/connectors/building-connectors/developing-source-connectors.mdx

```diff
@@ -74,128 +72,117 @@ func (s *Source) Parameters() map[string]sdk.Parameter {
 
 Validates and stores configuration data for the connector. Any complex validation logic should be implemented here.
 
-  ```go
-  func (s *Source) Configure(ctx context.Context, cfg map[string]string) error {
-  sdk.Logger(ctx).Info().Msg("Configuring Source...")
-  err := sdk.Util.ParseConfig(cfg, &s.config)
-  if err != nil {
-      return fmt.Errorf("invalid config: %w", err)
-  }
-  // add custom validations here
-  return nil
-  }
-  ```
+```go
+func (s *Source) Configure(ctx context.Context, cfg config.Config) error {
+    err := sdk.Util.ParseConfig(ctx, cfg, &s.config, NewSource().Parameters())
+    if err != nil {
+        return err
+    }
+    // add custom validations here
+    return nil
+}
+```
 
 ### `Open()`
 
 Prepares the connector to start producing records based on the last known successful position. If needed, the connector should open connections in this function.
 
-    ```go
-    func (s *Source) Open(ctx context.Context, pos sdk.Position) error {
-        configDirPath := s.config.Directory
-        files, err := ioutil.ReadDir(configDirPath)
-        if err != nil {
-            return fmt.Errorf("error reading directory '%s': %w", configDirPath, err)
-        }
+Every record read by a source connector has
+a [position](https://conduit.io/docs/features/opencdc-record#fields) attached.
+The position given to `Open()` is the position of the record that was last read
```

Comment:
Instead of "last read" I would say it's the last one that was successfully processed end-to-end, to make it clear that it's the _next_ record after this one that should be produced by the source.

Reply:
Thanks, I'll change that now.